### PR TITLE
Use new quimb TN string format

### PR DESCRIPTION
**Pull request by 95-martin-orion**
Resolves the "tests are failing" part of #5524.

### CODE REVIEW

**Review by viathor — APPROVED** [review]
AFAIR, this is the first time this happened with quimb. I'd consider pinning if it keeps happening and affects us more seriously.

**Comment by 95-martin-orion** [comment]
> This change will cause tests to fail for people who have older versions of quimb installed locally, which seems less than ideal.

This isn't the first time we've done something like this - the usual solution is to announce the change and recommend re-installing Cirq requirements.

> I don't think we should be relying on string representations provided by third-party libraries to be stable in our tests.

This is the cost of allowing MPS to use quimb-based state for its string representation. Some alternatives:
- Manually construct a string for the quimb state in Cirq
- Don't reveal the state in the string representation
- Don't test the string representation (likely fails coverage checks)

@maffoo, what are your thoughts? I'll hold off on merging until I have your approval.

**Pull request merged**
